src/main/java/com/uid2/admin/vertx/service/ServiceLinkService.java

```diff
@@ -118,6 +128,102 @@ private void handleServiceLinkAdd(RoutingContext rc) {
         }
     }
 
+    private void handleServiceLinkUpdate(RoutingContext rc) {
+        try {
+            siteProvider.loadContent();
+            serviceProvider.loadContent();
+
+            JsonObject body = rc.body().asJsonObject();
+            if (body == null) {
+                ResponseUtil.error(rc, 400, "json payload required but not provided");
+                return;
+            }
+            String linkId = body.getString("link_id");
+            Integer serviceId = body.getInteger("service_id");
+            Integer siteId = body.getInteger("site_id");
+            String name = body.getString("name");
+            if (siteId == null || serviceId == null) {
+                ResponseUtil.error(rc, 400, "required parameters: site_id, service_id");
+                return;
+            }
+
+            if (serviceProvider.getService(serviceId) == null) {
+                ResponseUtil.error(rc, 404, "service_id " + serviceId + " not valid");
+                return;
+            }
+
+            if (siteProvider.getSite(siteId) == null) {
+                ResponseUtil.error(rc, 404, "site_id " + siteId + " not valid");
+                return;
+            }
+
+            ServiceLink serviceLink = this.serviceLinkProvider.getAllServiceLinks()
+                    .stream().filter(s -> s.getServiceId() == serviceId && s.getSiteId() == siteId)
+                    .findFirst()
```

Comment:
If we are allowing multiple service_links to a service, will this be enough to identify the service_link

Reply:
Good catch - update to include the name, so the only thing you can change is the LinkId. As we can delete a ServiceLink, it is easy to recreate it if we need to.

Reply:
Having a bunch of disabled keys that we know are wrong doesn't seem like a good idea. Having the ability to edit, as well as delete, gives us the option to maintain the keys. We currently don't have the concept of disabled service_links either.